contracts/vaults/BaseVaultWithSwap/RibbonThetaVaultWithSwap.sol

```diff
@@ -501,4 +510,17 @@ contract RibbonThetaVaultWithSwap is RibbonVault, RibbonThetaVaultStorage {
             uint256(vaultState.lockedAmount).sub(unlockedAssetAmount)
         );
     }
+
+    /**
+     * @notice pause a user's vault position
+     */
+    function pausePosition() external {
+        address _vaultPauser = vaultPauser;
+        require(_vaultPauser != address(0)); // Removed revert msgs due to contract size limit
+        _redeem(0, true);
+        uint256 heldByAccount = balanceOf(msg.sender);
+        _approve(msg.sender, _vaultPauser, heldByAccount);
+        _transfer(msg.sender, _vaultPauser, heldByAccount);
```

Comment:
We shouldn't be transferring the vault shares from the vault itself, this has to be done within the VaultPauser contract with `transferFrom`